web/pages/review/index.tsx

```diff
@@ -3,42 +3,63 @@ import PageSection, { SectionArticle } from "@components/PageSection";
 import ProtectedRoute from "@components/ProtectedRoute";
 import { Flashcard } from "@src/types/Flashcard";
 import { getFlashcards } from "@src/getFlashcards";
-import { useAuth } from "react-oidc-context";
+import { AuthContextProps, useAuth } from "react-oidc-context";
 import FolderListView from "@components/FolderListView";
 import { useState } from "react";
 import ReviewPopup from "@components/routeReview/ReviewPopup";
 import { useQuery } from "@tanstack/react-query";
 import getRootFolder from "@src/getRootFolder";
 import Button from "@components/Button";
+import {
+    reviewFidsKey,
+    reviewModesKey,
+    useSessionStorage,
+} from "@src/useStorage";
 import { useRouter } from "next/router";
-import { getRequest } from "@src/apiRequest";
-import { ParsedUrlQuery } from "querystring";
-import { NextFlashcardGetResp } from "@src/types/NextFlashcardGetResp";
-import { Mode } from "@src/types/Mode";
+/*
+This page allows the user to select flashcards from the list (displayed by 
+selecting folders) and modes from the popup to use in a review session.
+These will store in sessionStorage and Router will redirect to review/cards/
+*/
 
-interface ReviewQueryParams extends ParsedUrlQuery {
-    modes: Mode[];
-    flashcardIds: string[];
+function calcMaxWidth(flashcardIds: number[]) {
+    // deprecated
+    const flashcardIdLength = Math.max(...flashcardIds).toString().length;
+    return Math.max(6, flashcardIdLength * 2 + 2);
 }
 
-function calcMaxWidth(fIds: number[]) {
-    const fidLength = Math.max(...fIds).toString().length;
-    return Math.max(6, fidLength * 2 + 2);
+async function getAccessedFlashcards(
+    auth: AuthContextProps,
+    selectedFolders: number[]
+) {
+    let flashcards: Flashcard[] = [];
+    if (!auth.user?.id_token) return flashcards;
+    // TODO: this is inefficient. it should be done in one request, or make better
+    // usage of QueryCache
+    for (const folderId of selectedFolders) {
+        flashcards = flashcards.concat(
+            await getFlashcards(auth.user.id_token, folderId)
+        );
+    }
+    return flashcards;
 }
 
 export default function ReviewIndex() {
     const auth = useAuth();
+    const router = useRouter();
+    // breaking setState convention to indicate storage?
```

Comment:
Wdym?

Reply:
Our use...Storage hook uses useState under the hood, but my naming convention didn't conform to the common pattern `[state, setState]`, which I justified with the above comment. In hindsight this was probably superfluous unless you object to it.